src/Hydrator/MoneyHydrator.php

```diff
@@ -20,7 +20,7 @@ class MoneyHydrator implements HydratorInterface
     public function extract($object)
     {
         return [
-            'amount' => $object->getAmount(),
+            'amount' => $object->getAmount() / 100,
```

Comment:
A user will always enter "1.25 USD". We have to have 125 as amount value. So putting this behaviour in a custom form element, which translates (in zf2 terms: filters) the input (1.25) to 125. The hydrator then does not need to divide by 100. Good catch @danizord !

About the closureToPhp and closureToMongo methods: They are indeed needed for doctrine odm mongodb adapter to work. They do not exist in doctrine orm.
We tested the current mongo implementation (which I contributed to, too), but the automatic string casting does not happen. Additionally we have to concider the null checks. So this in needed in order to work cleanly.